lib/datasource/go/index.ts

```diff
@@ -64,7 +66,32 @@ async function getDatasource(goModule: string): Promise<DataSource | null> {
       };
     }
   } else {
-    logger.trace({ goModule }, 'No go-source header found');
+    // GitHub Enterprise only returns a go-import meta
+    const importMatch = regEx(
+      `<meta\\s+name="go-import"\\s+content="([^\\s]+)\\s+([^\\s]+)\\s+([^\\s]+)">`
+    ).exec(res);
+    if (importMatch) {
+      const [, prefix, , goImportURL] = importMatch;
+      if (!goModule.startsWith(prefix)) {
+        logger.trace({ goModule }, 'go-import header prefix not match');
+        return null;
+      }
+      logger.debug({ goModule, goImportURL }, 'Go lookup import url');
+
+      // get server base url from import url
+      const httpMatch = httpRegExp.exec(goImportURL);
+
+      const split = goModule.split('/');
+      const lookupName = split[1] + '/' + split[2];
+
+      return {
+        datasource: github.id,
```

Comment:
`go-import` can be used for non-GitHub hosts too though right?

Reply:
It took me quite awhile to work out the difference between `go-import` and `go-source`.

- `go-source` was implement as part of gddo (godoc.org) but was never made part of core Go
- `go-import` is used by `go get` and other Go tooling:
> If the import path is not a known code hosting site and also lacks a version control qualifier, the go tool attempts to fetch the import over https/http and looks for a <meta> tag in the document's HTML <head>.
>The meta tag has the form:
> `<meta name="go-import" content="import-prefix vcs repo-root">`

https://golang.org/cmd/go/#hdr-Remote_import_paths

So the answer to your question is yes - `go-import` must be specified by any code hosting site that serves a Go package otherwise it won't work (Go does have site specific behaviour but that's generally stuff like github.com and bitbucket which both implement the `go-import` header anyway)

Reply:
Based on your research, would you recommend in future that we change to use the `go-import` header first instead of the `go-source` header today? It seems we only fall back go `go-import` if we can't find `go-source`

Reply:
Definitely, go-import is the standard way (as defined in the official dependency manager anyway) for following imports where as I believe go-source is specifically used for godoc following. As well as GHE and I think eventually GitHub.com not having go-source set at all.

Reply:
Also these headers can be specified on any host and assuming anything that exposes these is either GitHub / GHE or GitLab based on the combination of these and URL matching against gitlab* or github.com is probably quite naive, but most likely out of scope for this PR at least. In future it would maybe be best to detect based off a Server header returning GitHub or GitLab.

Another thing I experimented with was the Git datasource provider to read tagged releases there but I didn't get much luck with it.